SoftwareDesign.md

```diff
@@ -0,0 +1,6 @@
+### Software Design
+1. Favorite Design Pattern
+> Of the few I've had time to study this week, the `lazy initialization` has to be my favorite. Being able to delay creation of an object, calculation of a value, or any other expensive process just seems too good to be true. From a database standpoint, this would be a great way to create a database connection, but only at the point you need to retrieve data from that database. In a larger application, this allows the user to install or get only what they need as they need it, instead of having to wait for all features to install up front.
+
+2. Anti-pattern
```

Comment:
It's good that you are recognizing this habit early on, as we are all guilty of it at some point.